aws/resource_aws_appstream_stack.go

```diff
@@ -0,0 +1,570 @@
+			return diag.FromErr(fmt.Errorf("error setting `%s` for AppStream Stack (%s): %w", "tags_all", d.Id(), err))
+		}
+	}
+	return nil
+}
+
+func resourceAwsAppStreamStackUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*AWSClient).appstreamconn
+
+	input := &appstream.UpdateStackInput{
+		Name: aws.String(d.Id()),
+	}
+
+	if d.HasChange("access_endpoints") {
+		input.AccessEndpoints = expandAccessEndpoints(d.Get("access_endpoints").(*schema.Set).List())
+	}
+
+	if d.HasChange("application_settings") {
+		input.ApplicationSettings = expandApplicationSettings(d.Get("application_settings").(*schema.Set).List())
+	}
+
+	if d.HasChange("description") {
+		input.Description = aws.String(d.Get("description").(string))
+	}
+
+	if d.HasChange("display_name") {
+		input.DisplayName = aws.String(d.Get("display_name").(string))
+	}
+
+	if d.HasChange("feedback_url") {
+		input.FeedbackURL = aws.String(d.Get("feedback_url").(string))
+	}
+
+	if d.HasChange("redirect_url") {
+		input.RedirectURL = aws.String(d.Get("redirect_url").(string))
+	}
+
+	if d.HasChange("user_settings") {
+		input.UserSettings = expandUserSettings(d.Get("user_settings").([]interface{}))
+	}
+
+	resp, err := conn.UpdateStack(input)
+
+	if err != nil {
+		diag.FromErr(fmt.Errorf("error updating Appstream Stack (%s): %w", d.Id(), err))
+	}
+
+	if d.HasChange("tags") {
+		arn := aws.StringValue(resp.Stack.Arn)
+
+		o, n := d.GetChange("tags")
+		if err := keyvaluetags.AppstreamUpdateTags(conn, arn, o, n); err != nil {
+			return diag.FromErr(fmt.Errorf("error updating Appstream Stack tags (%s): %w", d.Id(), err))
+		}
+	}
+
+	return resourceAwsAppStreamStackRead(ctx, d, meta)
+}
+
+func resourceAwsAppStreamStackDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*AWSClient).appstreamconn
+
+	_, err := conn.DeleteStackWithContext(ctx, &appstream.DeleteStackInput{
+		Name: aws.String(d.Id()),
+	})
+	if err != nil {
+		if tfawserr.ErrCodeEquals(err, appstream.ErrCodeResourceNotFoundException) {
+			return nil
+		}
+		return diag.FromErr(fmt.Errorf("error deleting Appstream Stack (%s): %w", d.Id(), err))
+	}
+
+	// Will wait to finish to delete because after delete it makes a stack inactive then it deletes
+	time.Sleep(waiter.StackSleep)
+
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("error reading Appstream Stack (%s): %w", d.Id(), err))
+	}
+
+	return nil
+}
+
+func expandAccessEndpoints(tfList []interface{}) []*appstream.AccessEndpoint {
+	if len(tfList) == 0 {
+		return nil
+	}
+
+	var apiObjects []*appstream.AccessEndpoint
+
+	for _, v := range tfList {
+		v1 := v.(map[string]interface{})
+
+		endpoint := &appstream.AccessEndpoint{
+			EndpointType: aws.String(v1["endpoint_type"].(string)),
+		}
+		if v2, ok := v1["vpce_id"]; ok {
+			endpoint.VpceId = aws.String(v2.(string))
+		}
+
+		apiObjects = append(apiObjects, endpoint)
+	}
```

Comment:
```suggestion
	for _, tfMapRaw := range tfList {
		tfMap, ok := tfMapRaw.(map[string]interface{})
		
		if !ok {
			continue
		}
		
		apiObject := expandAccessEndpoint(tfMap)
		
		if apiObject == nil {
			continue
		}

		apiObjects = append(apiObjects, apiObject)
	}
```

(You'll need to write `expandAccessEndpoint()`.)

Breaking this into two functions helps in a few ways:
1. helps catch some unusual problems
2. there are times where the singular function comes it handy by itself
2. makes it easier to swap out for a generic solution down the road